### PR TITLE
Add tests for build command failures, fix for Questa/Modelsim

### PR DESCRIPTION
**Pull request by jonpovey**

### CODE REVIEW

**Comment by jonpovey** [comment]
I'll take a look at the failures later. Unexpected passing tests for icarus, think it is skipping compilation when it should not.

**Comment by themperek** [comment]
Can you possibly make 2 sperate PRs for this? 
Test are failing (Windows has other issues but the rest should be ok).

File: cocotb_test/simulator.py [hunk 1]
```diff
@@ -268,7 +270,7 @@ def execute(self, cmds):
             self.process = None
             if p.returncode:
                 self.logger.error("Command terminated with error %d" % p.returncode)
-                return
+                raise CommandFailedError()
```
Comment (themperek) [inline cocotb_test/simulator.py:273]: Maybe better?
```python3
raise SystemExit("Process '%s' termindated with error %d" % (p.args[0], p.returncode))
```

**Comment by jonpovey** [comment]
> Can you possibly make 2 sperate PRs for this? Test are failing (Windows has other issues but the rest should be ok).

OK, I assume you mean one with the tests and another with the Modelsim/Questa stuff. I'll change this one into just the tests.

**Comment by jonpovey** [comment]
Created #165 and #166 with the exception change suggested. This PR is obsolete

**Pull request closed without merging**